tests/unittests/test_app_ids_configurable.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+from unittest import mock
+from argparse import Namespace
+from tap_pendo import main
+
+
+class TestAppIdConfiguration(unittest.TestCase):
+    @mock.patch("tap_pendo.utils.parse_args", side_effect=lambda required_config_keys: Namespace(config={"start_date": "", "x_pendo_integration_key": "", "period":""},  discover=True))
+    @mock.patch("tap_pendo.discover_streams", side_effect=lambda config: {})
+    def test_app_ids_not_in_config(self, mocked_discover_stream, mocked_parse_args):
+        """
+            To verify that if app_ids is not in configure file then select all apps and run discover mode.
+        """
+        main()
```

Comment:
Add at least one assertion about app_ids value in config or any other relevant assertion.